base_import_async/models/base_import_import.py

```diff
@@ -117,34 +91,37 @@ def _create_csv_attachment(self, fields, data, options, file_name):
                             delimiter=str(options.get(OPT_SEPARATOR)),
                             quotechar=str(options.get(OPT_QUOTING)))
         encoding = options.get(OPT_ENCODING, 'utf-8')
-        writer.writerow(_encode(fields, encoding))
+        writer.writerow(fields)
         for row in data:
-            writer.writerow(_encode(row, encoding))
+            writer.writerow(row)
         # create attachment
+        datas = encodebytes(f.getvalue().encode(encoding))
         attachment = self.env['ir.attachment'].create({
             'name': file_name,
-            'datas': f.getvalue().encode('base64'),
+            'datas': datas,
             'datas_fname': file_name
         })
         return attachment.id
 
     @api.model
     def _read_csv_attachment(self, att_id, options):
         att = self.env['ir.attachment'].browse(att_id)
-        f = StringIO(att.datas.decode('base64'))
+        decoded_datas = decodebytes(att.datas)
+        encoding = options.get(OPT_ENCODING, 'utf-8')
+        f = StringIO(decoded_datas.decode(encoding))
```

Comment:
`f = io.TextIOWrapper(io.BytesIO(decoded_datas), encoding="utf8")` should be more memory efficient by decoding on the fly.